src/Vlingo.Actors/Plugin/Mailbox/AgronaMPSCArrayQueue/ManyToOneConcurrentArrayQueueDispatcher.cs

```diff
@@ -32,27 +33,36 @@ public class ManyToOneConcurrentArrayQueueDispatcher : IRunnable, IDispatcher
             Mailbox = new ManyToOneConcurrentArrayQueueMailbox(this, mailboxSize, totalSendRetries);
             this.throttlingCount = throttlingCount;
             closed = new AtomicBoolean(false);
-            cancellationTokenSource = new CancellationTokenSource();
+            dispatcherTokenSource = new CancellationTokenSource();
+            backoffTokenSource = CancellationTokenSource.CreateLinkedTokenSource(dispatcherTokenSource.Token);
         }
 
-        public void Close() => closed.Set(true);
+        public void Close()
+        {
+            closed.Set(true);
+            dispatcherTokenSource.Cancel();
+            dispatcherTokenSource.Dispose();
+            backoffTokenSource.Dispose();
```

Comment:
If this dispatcher was waiting in the `backoff.Now(...)` call, disposing the backoff token will raise a different exception that you haven't handled. Or, should you cancel the token first?

Reply:
This will raise the cancel exception first because the parent `dispatcherTokenSource` token source is cancelled first so it cascades to `backoffTokenSource`. Cancellation exception will throw before dispose is even handled

Reply:
My bad, I missed the fact that backoff token is linked with the dispatcher token. All good.